package.json

```diff
@@ -40,7 +40,8 @@
     "*.toml": "pnpm run format:toml"
   },
   "engines": {
-    "pnpm": ">=7.25.0"
+    "pnpm": "8.2.0",
```

Comment:
I'd like to suggest using `>= 8.0.0 <= 8.2.0`(The syntax might be wrong, but you get the idea).

As long as we are using pnpm with the same lockfile-version, we don't need a fixed version here.

If a developer happens to install pnpm@8.1.0 already, there's no need to forbid him from executing `pnpm install` and force him to reinstall pnpm@8.2.0.

Reply:
Hi @hyf0 looking into it does not seem to be possible right now please check this issue : https://github.com/nodejs/corepack/issues/95
and the feature was actually removed with this PR: https://github.com/nodejs/corepack/pull/18 please check the reasons behind explaining the potential side effects not specifying an exact version
Currently allowed syntax is:
```
String does not match the pattern of "(npm|pnpm|yarn)@\d+\.\d+\.\d+(-.+)?".
```
So i would guess having a strict version there would be the way to go.
To make it smooth for dev experience I would suggest to add a preinstall step as the following:
```
 "preinstall": "node scripts/meta/setup.mjs",
 ```
 This would automatically set the right version of the package manager on `pnpm install command`
 ```
root@codespaces-569785:/workspaces/rspack# pnpm install
npm/darwin-arm64                         |  WARN  Unsupported platform: wanted: {"cpu":["arm64"],"os":["darwin"],"libc":["any"]} (current: {"os":"linux","cpu":"x64","libc":"glibc"})
npm/darwin-x64                           |  WARN  Unsupported platform: wanted: {"cpu":["x64"],"os":["darwin"],"libc":["any"]} (current: {"os":"linux","cpu":"x64","libc":"glibc"})
npm/win32-x64-msvc                       |  WARN  Unsupported platform: wanted: {"cpu":["x64"],"os":["win32"],"libc":["any"]} (current: {"os":"linux","cpu":"x64","libc":"glibc"})
Scope: all 51 workspace projects
. pnpm:devPreinstall$ is-ci || node ./scripts/doctor.mjs
│ Yeah. Doctor here.
└─ Done in 89ms
Lockfile is up to date, resolution step is skipped
Packages: +2160
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
node_modules/@nestjs/core: Running postinstall script, done in 236ms
node_modules/@swc/core: Running postinstall script, done in 89ms
node_modules/core-js: Running postinstall script, done in 116ms
node_modules/core-js-pure: Running postinstall script, done in 136ms
node_modules/esbuild: Running postinstall script, done in 149ms
node_modules/playwright-chromium: Running install script, done in 304ms
node_modules/puppeteer: Running postinstall script, done in 225ms
node_modules/svelte-check/node_modules/svelte-preprocess: Running postinstall script, done in 14ms
node_modules/svelte-preprocess: Running postinstall script, done in 12ms
packages/rspack-plugin-minify/node_modules/esbuild: Running postinstall script, done in 62ms
node_modules/styled-components: Running postinstall script, done in 57ms
packages/rspack prepare$ pnpm precompile-schema
│ > @rspack/core@0.1.7 precompile-schema /workspaces/rspack/packages/rspack
│ > node ./scripts/precompile-schema.js
└─ Done in 722ms
. preinstall$ node scripts/meta/setup.mjs
│ ⏺️ Running `corepack enable`
│ ⏹️ Finish  `corepack enable` with 0
│ (node:21103) ExperimentalWarning: Importing JSON modules is an experimental feature. This feature could change at any time
│ (Use `node --trace-warnings ...` to show where the warning was created)
└─ Done in 134ms
. prepare$ is-ci || husky install
│ husky - Git hooks installed
└─ Done in 91ms
The integrity of 3986 files was checked. This might have caused installation to take longer.
Done in 5.8s
root@codespaces-569785:/workspaces/rspack# 
```